InnoGotchi.API/InnoGotchi.API/Extensions/ServiceExtensions.cs

```diff
@@ -0,0 +1,131 @@
+﻿using AutoMapper;
+using Infrastructure.Persistance;
+using Infrastructure.Persistance.Repositories;
+using Infrastructure.Services;
+using InnoGotchi.API.Filters.ActionFilters;
+using InnoGotchi.Application.Contracts.Repositories;
+using InnoGotchi.Application.Contracts.Services;
+using InnoGotchi.Application.Mapper;
+using InnoGotchi.Domain.Enums;
+using Microsoft.AspNetCore.Authentication.JwtBearer;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.IdentityModel.Tokens;
+using Microsoft.OpenApi.Models;
+using System.Text;
+
+namespace InnoGotchi.API.Extensions
+{
+    public static class ServiceExtensions
+    {
+        public static void ConfigureSqlContext(this IServiceCollection services,
+            IConfiguration configuration) =>
+                services.AddDbContext<AppDbContext>(opts =>
+                    opts.UseSqlServer(configuration.GetConnectionString("sqlConnection"), b =>
+                        b.MigrationsAssembly("Infrastructure.Persistance")));
+
+        public static void ConfigureRepositoryManager(this IServiceCollection services) =>
+            services.AddScoped<IRepositoryManager, RepositoryManager>();
+
+        public static void ConfigureAutoMapper(this IServiceCollection services) =>
+            services.AddScoped(provider => new MapperConfiguration(mc =>
+            {
+                mc.AddProfile(new MappingProfile(provider.GetService<IPetConditionService>()));
+            }).CreateMapper());
+
+        public static void ConfigureServices(this IServiceCollection services)
+        {
+            services.AddScoped<IAuthenticationService, AuthenticationService>();
+            services.AddScoped<IFarmService, FarmService>();
+            services.AddScoped<IGenerateFarmStatisticsService, GenerateFarmStatisticsService>();
+            services.AddScoped<IGenerateTokenService, GenerateTokenService>();
+            services.AddScoped<IPetConditionService, PetConditionService>();
+            services.AddScoped<IPetService, PetService>();
+            services.AddScoped<IUserService, UserService>();
+            services.AddScoped<IAvatarService, AvatarService>();
+            services.AddScoped<ICheckUserService, CheckUserService>();
+        }
+
+        public static void ConfigurActionFilters(this IServiceCollection services)
+        {
+            services.AddScoped<ValidationFilterAttribute>();
+            services.AddScoped<CheckUserIdAttribute>();
+        }
+
+        public static void ConfigureJWT(this IServiceCollection services, IConfiguration configuration)
+        {
+            var jwtSettings = configuration.GetSection("JwtSettings");
```

Comment:
You should add null checks for the variables from appsettings / enviroment before using them. If such variable is null, then throw some meaningful exception

Reply:
made